tests/cmd/test_status.py

```diff
@@ -1,42 +1,60 @@
+import json
 from unittest import mock
 
 import pytest
+from pytest_voluptuous import Partial, S
+from voluptuous.validators import All, Invalid, Range
 
 from ggshield.cmd.main import cli
-from tests.conftest import assert_invoke_ok, my_vcr
-
-
-@pytest.mark.parametrize(
-    "cassette, json_output",
-    [
-        ("quota", True),
-        ("quota", False),
-        ("quota_half_remaining", False),
-        ("quota_low_remaining", False),
-    ],
-)
-def test_quota(cassette, json_output, snapshot, cli_fs_runner):
-    with my_vcr.use_cassette(cassette):
-        cmd = ["quota", "--json"] if json_output else ["quota"]
+from tests.conftest import assert_invoke_ok, must_match_pattern, my_vcr
+
+
+def test_quota(cli_fs_runner):
+    with my_vcr.use_cassette("quota"):
+        cmd = ["quota", "--json"]
         result = cli_fs_runner.invoke(cli, cmd, color=False)
         assert_invoke_ok(result)
-        snapshot.assert_match(result.output)
-
-
-@pytest.mark.parametrize(
-    "cassette, json_output",
-    [
-        ("test_health_check", True),
-        ("test_health_check", False),
-        ("test_health_check_error", False),
-    ],
-)
-def test_api_status(cassette, json_output, snapshot, cli_fs_runner):
-    with my_vcr.use_cassette(cassette):
-        cmd = ["api-status", "--json"] if json_output else ["api-status"]
+
+        def quota_values_must_match(output):
+            if output["count"] + output["remaining"] != output["limit"]:
+                raise Invalid("API calls count and remaining must sum to limit.")
+
+        assert S(
+            All(
+                Partial(  # Partial validation because of the "since" key
+                    {
+                        "count": All(int, Range(min=0, max=1000)),
+                        "limit": 1000,
+                        "remaining": All(int, Range(min=0, max=1000)),
```

Comment:
This fails on my machine, because my quotas are higher 😊. I think we can relax the tests here a bit by defining the 3 values as `All(int, Range(min=0))`. `quota_values_must_match` ensures everything remains consistent.